site/app/controllers/UserProfileController.php

```diff
@@ -100,14 +99,38 @@ public function changePassword() {
     }
 
 
+    /**
+     * @Route("/user_profile/change_pronouns", methods={"POST"})
+     * @return JsonResponse
+     * @throws \ImagickException
+     */
+    public function changePronouns() {
+        $user = $this->core->getUser();
+        $newPronouns = trim($_POST['pronouns']);
```

Comment:
Pronouns can be empty in case the user wants to reset them to be empty

Reply:
Sure.  You need to make sure the field `pronouns` exists in the `$_POST` array though, because it will otherwise throw an exception if the field is not found.  You probably check if the `pronouns` field exists, and set the pronouns to an empty string if not.

Reply:
fixed